service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -488,7 +488,9 @@ private void checkMessageLimit(
 
         final Promise<Long> result = Promise.promise();
         LOG.trace("running Prometheus query [URL: {}, query: {}]", url, query);
-        newQueryRequest(query).send(sendAttempt -> {
+        newQueryRequest(query)
+                .timeout(config.getQueryTimeout())
```

Comment:
IMHO this should also be done in the *newQueryRequest* method, shouldn't it? It should also only be set if the config value is > 0, right?

Reply:
Vertx web-client have already the logic for timeout when is zero or negative the timeout is disabled. But anyway,  I followed your suggestion and   I putted the `timeout`  in the `newQueryRequest `.